hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1251,6 +1234,9 @@ private boolean updateStorefiles(List<HStoreFile> sfs, long snapshotId) throws I
       // the lock.
       this.lock.writeLock().unlock();
     }
+    if (snapshotId > 0) {
+      this.memstore.clearSnapshot(snapshotId);
```

Comment:
Maybe we could add some comments here why we can clear snapshot outside the writeLock.

Reply:
Done.